### PR TITLE
Test Suite CentOS 6 support and fixes

### PR DESCRIPTION
**Pull request by dtrudg**
**Description of the Pull Request (PR):**

Support CentOS 6 for `make test` and travis rpmbuild tests. This is important now that we have integration of libs such as libarchive (used by docker-extract) that are quite different in older versions.

Changes:

- Let travis run test suite on CentOS 6
- Skip only python tests where python <2.7 (CentOS 6 has 2.6 and test code is not compatible)
- Skip arch linux container test if we are on a 2 series kernel (will get kernel too old on CentOS 6)
- Fix `SINGULARITY_OVERLAY_FS` check - this variable is always defined, 0 or 1 - so check for that, not `-n`
- Fix busybox non-overlay user bind test - `/opt` doesn't actually exist in the busybox container, so we have to user bind somewhere else to check functionality without overlay. Use `/var/tmp` which is available and empty.

**Checkoff for all PRs:**

- [X] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [X] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [X] I have tested this PR locally with a `make test`
- [X] This PR is NOT against the project's master branch
- [X] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [X] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by dctrud** [comment]
Close/re-open to re-trigger travis.

**Review by jmstover — APPROVED** [review]
This all looks good to me.

File: tests/32-action_options.sh [hunk 1]
```diff
@@ -54,7 +54,7 @@ stest 0 mkdir -p "$TESTDIR"
 stest 0 touch "$TESTDIR/testfile"
 stest 0 singularity exec --home "$TESTDIR" "$CONTAINER" test -f "$TESTDIR/testfile"
 stest 0 singularity exec --home "$TESTDIR:/home" "$CONTAINER" test -f "/home/testfile"
-if [ -n "${SINGULARITY_OVERLAY_FS:-}" ]; then
+if [  "$SINGULARITY_OVERLAY_FS" = "1" ]; then
```
Comment (GodloveD) [inline tests/32-action_options.sh:57]: Why this change?  Seems like the original line is more robust right?

**Review by GodloveD — COMMENTED** [review]
Just looking this over.  Have not tested yet.  One quick question about conditional test in `action_options.sh`

File: tests/32-action_options.sh (hunk 1 above)
Comment (dtrudg) [reply tests/32-action_options.sh:57]: The original line doesn't work..

> Fix SINGULARITY_OVERLAY_FS check - this variable is always defined, 0 or 1 - so check for that, not -n

In `test.sh.in` that variable is always getting defined to 0 or 1... so the original test for nonzero length always has the same outcome whether or not overlay is available. Here's the test.sh.in bit:

```bash
SINGULARITY_OVERLAY_FS="0"
KVERS=`uname -r`
if test -f "/lib/modules/$KVERS/modules.dep"; then
    if grep -q 'overlay.ko' "/lib/modules/$KVERS/modules.dep"; then
        SINGULARITY_OVERLAY_FS="1"
    fi
fi
```

**Comment by GodloveD** [comment]
@dctrud can you rebase this plz.  I'm getting this:
```
configure: error: Unable to find the uuid headers, need package libuuid-devel (uuid-dev on Debian/Ubuntu)
```

**Comment by dtrudg** [comment]
@GodloveD - yup, next on my list.. working through it :-)

**Comment by dtrudg** [comment]
@GodloveD - this will be a by Monday if I have time thing. The travis setup did change a lot onto develpoment. May not be a 2.5.

File: tests/32-action_options.sh (hunk 1 above)
Comment (GodloveD) [reply tests/32-action_options.sh:57]: Ah.  Good catch.

**Comment by GodloveD** [comment]
OK.  PM me if you change your mind and want this reviewed.

**Comment by dtrudg** [comment]
@GodloveD - Dropping this from the 2.5 milestone as I'm not going to be able to get the travis stuff sorted by this PM. Noting that running test content on CentOS6 locally is important++ to ensure 2.x libarchive behaviour is good.

**Comment by GodloveD** [comment]
@dctrud any chance you could rebase this against release-2.5?

**Comment by dtrudg** [comment]
@GodloveD - will do my best on this - it's quite a big change from the old development to 2.5 for this, prob a good half-day or more messing as the build setup is a lot different (old development had the secure build stuff).

**Comment by dtrudg** [comment]
Closing this as out-of-date. Rebased PR #1621 is in progress.
